core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -3026,4 +3028,21 @@ private boolean isCTEDefinationTrait(RelNode rel) {
     }
     return false;
   }
+
+  private String adjustAliasForUpdateStatement(SqlNode node, RelNode rel, String currentAlias) {
+    if (isLogicalFilterWithSelectAndTableScan(node, rel)) {
+      return requireNonNull(((LogicalFilter) rel).getInput().getTraitSet()
+          .getTrait(TableAliasTraitDef.instance)).getTableAlias();
+    }
+    return currentAlias;
+  }
+
+  private boolean isLogicalFilterWithSelectAndTableScan(SqlNode node, RelNode rel) {
+    return node instanceof SqlSelect && rel instanceof LogicalFilter
```

Comment:
conditions can be modified as below
 ((LogicalFilter) rel).getInput().getTraitSet().size() > 2
        && "sstupdate".equals(
        requireNonNull(((LogicalFilter) rel).getInput()
            .getTraitSet().getTrait(TableAliasTraitDef.instance)).getStatementType())
&& node instanceof SqlSelect && rel instanceof LogicalFilter
        && ((LogicalFilter) rel).getInput() instanceof LogicalTableScan;

Reply:
we can not change this as first we need to check the node instance is of type logicalfilter.